src/renderer/utils/channel/request-from-channel.test.ts

```diff
@@ -57,64 +53,45 @@ describe("request from channel in renderer", () => {
       expect(actual).toBe("some-response");
     });
 
-    it("when invoking resolves with stringified string, resolves with string", async () => {
-      await invokeMock.resolve('"some-response"');
+    it("when resolving with number, resolves with number", async () => {
+      await invokeMock.resolve(10);
 
       const actual = await actualPromise;
 
-      expect(actual).toBe("some-response");
+      expect(actual).toBe(10);
     });
 
-    it("when invoking resolves with stringified boolean, resolves with boolean", async () => {
-      await invokeMock.resolve("true");
+    it("when resolving with number 0, resolves with number 0", async () => {
+      await invokeMock.resolve(0);
 
       const actual = await actualPromise;
 
-      expect(actual).toBe(true);
+      expect(actual).toBe(0);
     });
 
-    it("when invoking resolves with stringified number, resolves with number", async () => {
-      await invokeMock.resolve("42");
+    it("when resolving with true, resolves with true", async () => {
+      await invokeMock.resolve(true);
 
       const actual = await actualPromise;
 
-      expect(actual).toBe(42);
+      expect(actual).toBe(true);
     });
 
-    it("when invoking resolves with stringified object, resolves with object", async () => {
-      await invokeMock.resolve(JSON.stringify({ some: "object" }));
+    it("when resolving with false, resolves with false", async () => {
+      await invokeMock.resolve(false);
 
       const actual = await actualPromise;
 
-      expect(actual).toEqual({ some: "object" });
+      expect(actual).toBe(false);
     });
-  });
-
-  it("given string as message, when messaging to channel, sends stringified message", () => {
-    requestFromChannel(someChannel, "some-message");
-
-    expect(invokeMock).toHaveBeenCalledWith("some-channel-id", '"some-message"');
-  });
-
-  it("given boolean as message, when messaging to channel, sends stringified message", () => {
-    requestFromChannel(someChannel, true);
-
-    expect(invokeMock).toHaveBeenCalledWith("some-channel-id", "true");
-  });
 
-  it("given number as message, when messaging to channel, sends stringified message", () => {
-    requestFromChannel(someChannel, 42);
+    it("when resolving with object, resolves with object", async () => {
+      await invokeMock.resolve({ myField: true });
 
-    expect(invokeMock).toHaveBeenCalledWith("some-channel-id", "42");
-  });
-
-  it("given object as message, when messaging to channel, sends stringified message", () => {
-    requestFromChannel(someChannel, { some: "object" });
+      const actual = await actualPromise;
 
-    expect(invokeMock).toHaveBeenCalledWith(
-      "some-channel-id",
-      JSON.stringify({ some: "object" }),
```

Comment:
Hmm. I wonder if the serialisation and parsing is still being unit-tested.

Reply:
No because it is completely handled in our dependency.